packages/assets-controllers/src/TokensController.ts

```diff
@@ -598,16 +621,24 @@ export class TokensController extends BaseController<
    *
    * @param asset - The asset to be watched. For now only ERC20 tokens are accepted.
    * @param type - The asset type.
+   * @param interactingAddress - The address of the account that is requesting to watch the asset.
    * @returns Object containing a Promise resolving to the suggestedAsset address if accepted.
    */
-  async watchAsset(asset: Token, type: string): Promise<AssetSuggestionResult> {
-    const suggestedAssetMeta = {
+  async watchAsset(
+    asset: Token,
+    type: string,
+    interactingAddress?: string,
+  ): Promise<AssetSuggestionResult> {
+    const suggestedAssetMeta: SuggestedAssetMeta = {
       asset,
       id: this._generateRandomId(),
       status: SuggestedAssetStatus.pending as SuggestedAssetStatus.pending,
       time: Date.now(),
       type,
     };
+    if (interactingAddress) {
```

Comment:
TODO - Provide selectedAddress as fallback to ensure that suggestedAssetMeta always has an interactingAddress

Reply:
Hmm, what would the purpose of an interacting address be? I was thinking of the token list as being per-chain, not per chain/address. That seems to be how it's tracked in state at the moment as well. So there would be no way to watch an asset just for a single account.

Reply:
Oh I see, that's what this PR is for. Sorry I lost track of the context here for a moment.